go-runtime/compile/schema.go

```diff
@@ -863,13 +938,54 @@ func visitType(pctx *parseContext, pos token.Pos, tnode types.Type, isExported b
 		if underlying.String() == "any" {
 			return optional.Some[schema.Type](&schema.Any{Pos: goPosToSchemaPos(pos)})
 		}
+		if named, ok := tnode.(*types.Named); ok {
+			enumRef, doneWithVisit := visitEnumType(pctx, pos, named)
+			if doneWithVisit {
+				if enumRef != nil {
+					return optional.Some[schema.Type](enumRef)
+				}
+				return optional.None[schema.Type]()
+			}
+		}
 		return optional.None[schema.Type]()
 
 	default:
 		return optional.None[schema.Type]()
 	}
 }
 
+func visitEnumType(pctx *parseContext, pos token.Pos, named *types.Named) (schema.Type, bool) {
+	if pctx.enums[named.Obj().Name()] != nil {
+		return &schema.Ref{
+			Pos:    goPosToSchemaPos(pos),
+			Module: pctx.module.Name,
+			Name:   named.Obj().Name(),
+		}, true
+	}
+	if named.Obj().Pkg() == nil {
+		return nil, false
+	}
+	nodePath := named.Obj().Pkg().Path()
+	if !strings.HasPrefix(nodePath, pctx.pkg.PkgPath) {
+		// If this type is named and declared in another module, it's a reference.
+		// The only basic-typed references and non-any interface-typed references
+		// supported are enums.
+		if !strings.HasPrefix(named.Obj().Pkg().Path(), "ftl/") {
+			pctx.errors.add(noEndColumnErrorf(pos,
+				"unsupported external type %q", named.Obj().Pkg().Path()+"."+named.Obj().Name()))
+			return nil, true
+		}
+		base := path.Dir(pctx.pkg.PkgPath)
+		destModule := path.Base(strings.TrimPrefix(nodePath, base+"/"))
+		return &schema.Ref{
+			Pos:    goPosToSchemaPos(pos),
+			Module: destModule,
+			Name:   named.Obj().Name(),
+		}, true
+	}
+	return nil, false
```

Comment:
for consistency with the other visit functions, can we return an optional here instead of a type + bool? then if we get a non-empty value back we can exit

Reply:
I had to add the bool to distinguish between the case where (1) we want to continue visiting other possible types vs (2) actually return None from `visitType()`, but. returning an option + bool would definitely make the wrapping a bit nicer. I'll go ahead and do that now.